### PR TITLE
add flags package

### PR DESCRIPTION
**Pull request by campoy**

### CODE REVIEW

**Comment by campoy** [comment]
PTAL @broady

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: go
+
+go:
+  - 1.5.4
+  - 1.6.2
+  - 1.7beta2
```
Comment (broady) [inline .travis.yml]: 1.7 please

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+language: go
+
+go:
+  - 1.5.4
+  - 1.6.2
```
Comment (broady) [inline .travis.yml]: 1.6.3

File: flags/color.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	rgba := (*color.RGBA)(c)
+	rgba.B, n = uint8(n%256), n/256
+	rgba.G, n = uint8(n%256), n/256
+	rgba.R = uint8(n % 256)
+	rgba.A = 255
+	return nil
+}
+
+// HexColor Int defines a hex color flag with specified name, default value, and usage string.
```
Comment (broady) [inline flags/color.go]: s/Int//

File: flags/color.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+// The return value is the address of an RGBA color variable that stores the value of the flag.
+func HexColor(name string, value color.Color, usage string) *color.RGBA {
```
Comment (broady) [inline flags/color.go]: Maybe return type should be *color.Color?

**Comment by broady** [comment]
Sexy flags

```
type hexColor struct {
    color.Color
}

func (c *hexColor) String() string {
    r, g, b, a := c.Color.RGBA()
    return fmt.Sprintf("rgba(%d, %d, %d, %d)", r, g, b, a)
}

func (c *hexColor) Set(s string) error {
    if len(s) != 6 {
        return fmt.Errorf("color should be 6 digits")
    }
    n, err := strconv.ParseInt(s, 16, 64)
    if err != nil {
        return fmt.Errorf("not hexadecimal: %v", err)
    }
    rgba := &color.RGBA{}
    rgba.B, n = uint8(n%256), n/256
    rgba.G, n = uint8(n%256), n/256
    rgba.R = uint8(n % 256)
    rgba.A = 255
    c.Color = rgba
    return nil
}

// HexColor defines a hex color flag with specified name, default value, and usage string.
// The return value is the address of an RGBA color variable that stores the value of the flag.
func HexColor(name string, value color.Color, usage string) color.Color {
    c := &hexColor{value}
    flag.Var(c, name, usage)
    return c
}

// HexColorVar defines a hex color flag with specified name, default value, and usage string.
// The argument c points to an RGBA color variable in which to store the value of the flag.
func HexColorVar(c *color.Color, name string, value color.Color, usage string) {
    p := &hexColor{value}
    *c = p
    flag.Var(p, name, usage)
}
```

File: .travis.yml (hunk 2 above)
Comment (campoy) [reply .travis.yml]: done

File: .travis.yml (hunk 1 above)
Comment (campoy) [reply .travis.yml]: done

File: flags/color.go (hunk 3 above)
Comment (campoy) [reply flags/color.go]: done

File: flags/color.go (hunk 4 above)
Comment (campoy) [reply flags/color.go]: done

File: flags/README.md [hunk 5]
```diff
@@ -7,6 +7,72 @@ consider to be useful.
...
+	"flag"
+	"fmt"
+	"image/color"
+
+	"github.com/campoy/tools/flags"
+)
+
+func main() {
+	a := flags.HexColor("a", color.White, "color a")
+	var b color.RGBA
```
Comment (broady) [inline flags/README.md]: var b color.Color

File: flags/README.md [hunk 6]
```diff
@@ -7,6 +7,72 @@ consider to be useful.
...
+	flags.HexColorVar(&b, "b", color.White, "color b")
+	flag.Parse()
+
+	fmt.Printf("a is %v; b is %v", *fg, bg)
```
Comment (broady) [inline flags/README.md]: *fg -> *a

bg -> b

File: flags/README.md (hunk 5 above)
Comment (campoy) [reply flags/README.md]: done

File: flags/README.md (hunk 6 above)
Comment (campoy) [reply flags/README.md]: done

**Comment by campoy** [comment]
The failing tests are unrelated to this package (I'll fix it in a different PR)

PTAL

**Comment by campoy** [comment]
ping @broady

File: flags/docs.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+// under the License is distributed on a "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package flags implements command-line flag parsing for types
+// that are not part of the standard library flag package, but I
+// consider to be useful.
+package flags
```
Comment (broady) [inline flags/docs.go:17]: filename: `doc.go`

File: flags/texttemplate.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+	"text/template"
+)
+
+type textTemplate struct {
+	text string
+	t    template.Template
+}
+
+func (t *textTemplate) String() string {
+	return fmt.Sprintf("%q", t.text)
```
Comment (broady) [inline flags/texttemplate.go:29]: why not just `return t.text`

File: flags/texttemplate.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func (t *textTemplate) Set(text string) error {
+	p, err := template.New("flag").Parse(text)
```
Comment (broady) [inline flags/texttemplate.go:33]: `New("")`?

File: flags/texttemplate.go [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	tt := &textTemplate{value, *t}
+	flag.Var(tt, name, usage)
+	return &tt.t
+}
+
+// TextTemplateVar defines a text template flag with specified name, default value, and usage string.
+// The argument t points to a text template variable in which to store the value of the flag.
+// If the given template fails to compile an error will be logged and the program will exit with
+// exit code 2. (os.Exit(2))
```
Comment (broady) [inline flags/texttemplate.go:60]: no need to do the (os.Exit(2))

**Comment by broady** [comment]
LGTM. doc comments could do with some tightening up, but i didn't look/comment in detail

**Pull request merged**